src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -1239,6 +1419,13 @@ private static void AssertOuterBuildAsNonRoot(ProjectGraphNode outerBuild, Dicti
 
                     innerBuild.ReferencingProjects.ShouldContain(outerBuildReferencer);
                     innerBuild.ReferencingProjects.ShouldNotContain(outerBuild);
+
+                    graph.TestOnly_Edges.TestOnly_HasEdge((outerBuild, innerBuild)).ShouldBeFalse();
```

Comment:
Wait, why no edge for outer->inner of the same project?

Reply:
These are outer builds that are not roots. For non-root outerbuilds, the project that references the outer build also references the inner builds.